internal/sidekick/internal/dart/annotate.go

```diff
@@ -365,9 +374,7 @@ func calculateDependencies(imports map[string]bool, constraints map[string]strin
 			}) {
 				constraint := constraints[name]
 				if len(constraint) == 0 {
-					// TODO(https://github.com/googleapis/librarian/issues/1989):
-					// Never emit "any" constraints.
-					constraint = "any"
+					return nil, fmt.Errorf("unknown version constraint for package %q (did you forget to add it to .sidekick.toml?)", name)
```

Comment:
+1, I like fast-failing here (letting people know some additional explicit configuration is necessary).